src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+                    val != JsonConstants.Tab)
+                {
+                    break;
+                }
+
+                if (val == JsonConstants.LineFeed)
+                {
+                    _lineNumber++;
+                    _lineBytePosition = 0;
+                }
+                else
+                {
+                    _lineBytePosition++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method contains the logic for processing the next value token and determining
+        /// what type of data it is.
+        /// </summary>
+        private bool ConsumeValue(byte marker)
+        {
+            if (marker == JsonConstants.Quote)
+                return ConsumeString();
+            else if (marker == JsonConstants.OpenBrace)
+                StartObject();
+            else if (marker == JsonConstants.OpenBracket)
+                StartArray();
+            else if ((uint)(marker - '0') <= '9' - '0' || marker == '-')
+                return ConsumeNumber();
+            else if (marker == 'f')
+                return ConsumeLiteral(JsonConstants.FalseValue, JsonTokenType.False);
+            else if (marker == 't')
+                return ConsumeLiteral(JsonConstants.TrueValue, JsonTokenType.True);
+            else if (marker == 'n')
+                return ConsumeLiteral(JsonConstants.NullValue, JsonTokenType.Null);
+            else
+            {
+                if (_readerOptions.CommentHandling != JsonCommentHandling.Default)
+                {
+                    if (_readerOptions.CommentHandling == JsonCommentHandling.AllowComments)
+                    {
+                        if (marker == JsonConstants.Solidus)
+                        {
+                            return ConsumeComment();
+                        }
+                    }
+                    else
+                    {
+                        // JsonCommentHandling.SkipComments
+                        if (marker == JsonConstants.Solidus)
+                        {
+                            if (SkipComment())
+                            {
+                                if (_consumed >= (uint)_buffer.Length)
+                                {
+                                    if (_isNotPrimitive && IsLastSpan)
+                                    {
+                                        if (_tokenType != JsonTokenType.EndArray && _tokenType != JsonTokenType.EndObject)
+                                            ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidEndOfJson);
+                                    }
+                                    return false;
+                                }
+
+                                byte first = _buffer[_consumed];
+
+                                if (first <= JsonConstants.Space)
+                                {
+                                    SkipWhiteSpace();
```

Comment:
If I remember what SkipWhiteSpace does correctly, if `first` is BEL (U+0007) then this will enter SkipWhiteSpace, then SkipWhiteSpace will throw.

It'd be nice to see all the places checking `<= Space` have a comment along the lines of "All legal whitespace characters in ECMA-404 2nd edition are in the 0x00 - 0x20 range, and SkipWhiteSpace will reject the invalid ones."

Partially it's a way to keep track of things if 3rd edition comes along and says that all of Unicode Whitespace is legal, and partially it's a way to stop me from thinking "uh, where in this flow did extraneous EOT/NUL/BEL/etc get handled?"